_posts/2019-07-09-inception-pipelines-pt7.markdown

```diff
@@ -0,0 +1,171 @@
+
+1. Checkout the code [from here](https://github.com/MechanicalRock/InceptionPipeline/tree/post/part-7).
+2. Change all the Inception Pipeline values in the `aws_seed.json` and `aws_seed-cli-parameters.json` files to your specific values. See the [original post](https://mechanicalrock.github.io/2018/03/01/inception-pipelines-pt1.html) if you need a refresher on how to do this.
+3. Open the `infrastructure/capability-global-cloudtrail.json` and set the following values:
+
+    | Name                                          | Description                                                                                                                                      |
+    | --------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
+    | ParamBucketPrefix                             | A prefix for the bucket used to store the CloudTrail logs. A value of `-cloudtrail` is appended to this name to form the complete bucket name.   |
+    | ParamKmsKeyAlias                              | A human friendly alias for the KMS key used to protect the files stored in S3.                                                                   |
+    | ParamMasterAccount                            | The 12-digit AWS Account ID of your AWS Organizations master account. It is used to grant the appropriate levels of access between the accounts. |
+    | ParamTransitionLogsToGlacierAfterThisManyDays | To save on storage costs, move files older than this many days to Glacier storage. Defaults to 14 days.                                          |
+    | ParamExpireLogsFromGlacierAfterThisManyDays   | To save on storage costs, delete files older than this many days from Glacier. Defaults to approximately 6 months (180 days).                    |
+
+4. Deploy the pipeline and wait for it to complete. Once it has you will need to extract the following values from the infrastructure stack. These will be used when configuring the CloudTrail trail in the next section:
+
+    | Name       | Description                                                       |
+    | ---------- | ----------------------------------------------------------------- |
+    | BucketName | The complete name of the S3 bucket where we are storing the logs. |
+    | KmsKeyArn  | The KMS Key ARN to use to encrypt the logs.                       |
+
+## Enable AWS Organizations Global CloudTrail
+
+Now that the underlying infrastructure is deployed, we can start on ~~world domination~~ configuring Global [CloudTrail](https://aws.amazon.com/cloudtrail/)! Since this is a once-off activity it is easier and quicker to deploy via the console; one of the ultra-rare times that I will use the console for deployments.
```

Comment:
Is there a CLI alternative?

Reply:
There is a CLI alternative; however for this post I chose to focus on the UI approach.